File: test/fibonacci.c
#include <stdio.h>
#include <time.h>
/**
 * 斐波那契数列 递归测试
 * 40 403ms
 * 45 4400s
 */
unsigned long fibonacci(int num)
{
  if (num > 2)
  {
    return fibonacci(num - 1) + fibonacci(num - 2);
  }
  else
  {
    return 1;
  }
}

main()
{
  int fib = 40;
  clock_t start, end;
  unsigned long ret;
  start = clock();
  ret = fibonacci(fib);
  end = clock();

  printf("fib %d = %lu\n", fib, ret);
  printf("time = %ld\n", end - start);
}

File: go/fib/fibonacci.go
package main

import (
	"fmt"
	"time"
)

/**
 * 斐波那数列 递归测试
 * 40 420.6252ms
 * 45 4.6168056s
 */

func main() {
	var fib int = 45
	start := time.Now()
	var ret int = fibonacci(fib)
	end := time.Since(start)

	fmt.Printf("fib %v = %v\n", fib, ret)
	fmt.Println("time = ", end)
}

func fibonacci(num int) int {
	if num <= 2 {
		return 1
	}
	return fibonacci(num-1) + fibonacci(num-2)
}
